specs/Print.md

```diff
@@ -0,0 +1,473 @@
+/// The printable pdf data is returned in the `pdfStream` object.
+[uuid(78A9FC37-249B-4E7E-A2E6-20CAAA14AE7B), object, pointer_default(unique)]
+interface ICoreWebView2PrintToPdfStreamCompletedHandler : IUnknown {
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] IStream* pdfStream);
+}
+
+/// Settings used by the `PrintWithSettings` method.
+[uuid(DA3AD25C-D121-44DC-83DA-BF5E50DBF0DF), object, pointer_default(unique)]
+interface ICoreWebView2PrintSettings2 : ICoreWebView2PrintSettings {
+  /// Page range to print. Defaults to empty string, which means print all pages.
+  ///
+  /// The PageRanges property is a list of page ranges specifying one or more pages that should be printed separated by commas. Any whitespace between page ranges is ignored.
+  /// A valid page range is either a single integer identifying the page to print, or a range in the form `[start page]-[last page]` where `start page` and `last page` are integers identifying the first and last inclusive pages respectively to print. Every page identifier is an integer
+  /// greater than 0 unless wildcards are used (see below examples). The first page is 1.
+  ///
+  /// In a page range of the form `[start page]-[last page]` the start page number must be larger than 0 and less than or equal to the document's total page count.
+  /// If the `start page` is not present, then 1 is used as the `start page`.
+  /// The `last page` must be larger than the `start page`.
+  /// If the `last page` is not present, then the document total page count is used as the `last page`.
+  ///
+  /// If page range is not valid or if a page is greater than document total page count,
+  /// `ICoreWebView2PrintWithSettingsCompletedHandler` or ICoreWebView2PrintToPdfStreamCompletedHandler` handler will return `E_ABORT`.
+  ///
+  /// |       Example       | Valid |               Notes                   |
+  /// "2"                   |  Yes  | Assuming document total page count >= 2.
+  /// "1-4, 9, 3-6, 10, 11" |  Yes  | Assuming document total page count >= 11.
+  /// "1-4, -6"             |  Yes  | Assuming document total page count >= 6.
+  /// "2-"                  |  Yes  | Assuming document total page count >= 2, means from 2 to the * end.
+  /// "4-2, 11, -6"         |  No   |
+  ///  "-"                  |  Yes  | Assuming document total page count >= 1.
+  ///  "1-4dsf, 11"         |  No   | Regardless of document total page count.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`. See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings)
+  [propget] HRESULT PageRanges([out, retval] LPWSTR* value);
+
+  /// Set the `PageRanges` property.
+  [propput] HRESULT PageRanges([in] LPCWSTR value);
+
+  /// Number of pages per sheet. Minimum value is `1` and the maximum is `16`.
+  /// The default value is 1
+  [propget] HRESULT PagesPerSheet([out, retval] INT32* value);
```

Comment:
Must it be a power of 2? What happens if I ask for 5 pages per sheet? How are the pages arranged?

If I ask for two-sided printing, do I get PagesPerSheet pages on each side of the sheet? Or is it the total number of pages across both sides? E.g., if I set PagesPerSheet=2 and ask for two-sided printing, do I get one page on the front and another page on the back? Or do I get two pages on the front and two pages on the back?

Reply:
Looks like the format for the PagesPerSheet layout is 1, 2, 4, 6, 9, 16. I'll update it accordingly. 

It is pages per sheet even if two side printing is set.

If PagesPerSheet = 2 and TotalPageCount = 2, it prints two pages on the front side. 

If PagesPerSheet is 4 and TotalPageCount = 6, it prints four pages on the front side and 2 pages on the back side.

Reply:
Suppose I ask for two-sided printing, and I set this property to 4.

1. I get 2 pages on the front side and 2 pages on the back side (total 4 pages per sheet of paper).
2. I get 4 pages on the front side and 4 pages on the back side (total 8 pages per sheet of paper).

I think the behavior is (2), seeing as I can set this property to 9, which is not an even number.

Therefore, this property is really PagesPer**Side**, not pages per sheet.

Reply:
If there are 4 pages in a document and set two-side printing and this property is set as 4.

* It gives get 4 pages only on the front side and empty back side (total 4 pages only on front side).

If there are 8 pages in a document and set two-side printing and this property is set as 4.
 
* It gives 4 pages on the front side and 4 pages on the back side (total 8 pages front and back).